Makefile.am

```diff
@@ -456,6 +458,13 @@ CLAIM_PLUGIN_FILES = \
     claim/claim.h \
     $(NULL)
 
+ACLK_PLUGIN_FILES = \
```

Comment:
Those files shouldn't have to be compiled when they're not used, especially since there are still compile warnings. They should be fenced by a condition such as:
```
if ENABLE_ACLK
```

This affects the product in production.

Reply:
Removing all warnings :+1:

Reply:
Moved `../aclk/agent_cloud_link.h` inside common.h as `aclk/agent_cloud_link.h` near claim/claim.h

Reply:
Added condition to skip the compilation of the ACLK code entirely by default. It is enabled by setting env variable ACLK=yes